src/phongo_bson.c

```diff
@@ -162,6 +163,13 @@ void php_phongo_field_path_write_item_at_current_level(php_phongo_field_path* fi
 	php_phongo_field_path_ensure_allocation(field_path, field_path->size);
 
 	if (field_path->owns_elements) {
+		/* Note: owns_elements is only used for field paths parsed from a type
+		 * map, so it's unlikely that an element would already have been
+		 * allocated here. This is in contrast to BSON encoding/decoding, which
+		 * frequently updates the field path and does not own elements. */
+		if (UNEXPECTED(field_path->elements[field_path->size])) {
```

Comment:
This relies on the PHP macro from [zend_portability.h](https://github.com/php/php-src/blob/php-8.2.11/Zend/zend_portability.h#L362). Alternatively, I could have used [`BSON_UNLIKELY()`](https://github.com/mongodb/mongo-c-driver/blob/1.24.4/src/libbson/src/bson/bson-macros.h#L288).

Reply:
SGTM - I believe in the context of a PHP extension `UNEXPECTED` makes more sense than `BSON_UNLIKELY`.